### PR TITLE
[JN-393] Bulk populate enrollees

### PR DESCRIPTION
**Pull request by MatthewBemis**
Allows you to generate N random enrollees based off of the new `seed.json` profile. While enrollees will follow the same "shape" as the seed profile (allowing us to easily beef up which survey fields are filled out), some profile fields are randomly generated:

* Random given and family names
* Random birthday between 1920 and 2020
* Random username/contact email
* `doNotEmail` always set to true (we don't want to eat up sendgrid quota by accident)
* 5% chance of being withdrawn
* 95% chance of being consented

I also added a random `submittedHoursAgo` (within the past 20 days) to make these metrics charts more interesting:
Screenshot 2023-05-23 at 3 22 04 PM

TO TEST:

1) Load up the Admin swagger UI and try the new endpoint
2) Watch them show up!

### CODE REVIEW

File: populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java [hunk 1]
```diff
@@ -330,5 +333,25 @@ public Enrollee createNew(EnrolleePopDto popDto, StudyPopulateContext context, b
...
+    public void bulkPopulateEnrollees(String portalShortcode, EnvironmentName envName, String studyShortcode, List<String> usernamesToLink) {
+        StudyPopulateContext context = new StudyPopulateContext("portals/" + portalShortcode + "/studies/" + studyShortcode + "/enrollees/seed.json", portalShortcode, studyShortcode, envName, new HashMap<>());
+
+        usernamesToLink.forEach(username -> {
+            try {
+                String fileString = filePopulateService.readFile(context.getRootFileName(), context);
+                EnrolleePopDto popDto = objectMapper.readValue(fileString, getDtoClazz());
+                popDto.setLinkedUsername(username);
+                popDto.setConsented(PopulateUtils.randomBoolean(95)); //95% chance an enrollee will be consented
+                popDto.setWithdrawn(PopulateUtils.randomBoolean(5)); //5% chance an enrollee will be withdrawn
```
Comment (breilly2) [inline populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java:345]: And 0.25% chance that they'll be both not consented and withdrawn. Probably a very uncommon case indeed, if not impossible. Theoretically, we shouldn't have any data for someone who is not consented, so withdrawing shouldn't affect anything. Still a potentially interesting case, and one that I don't think I'd ever considered before.

File: populate/src/main/java/bio/terra/pearl/populate/service/PortalParticipantUserPopulator.java [hunk 2]
```diff
@@ -67,4 +74,45 @@ public PortalParticipantUser createPreserveExisting(PortalParticipantUser existi
     public PortalParticipantUser createNew(PortalParticipantUser popDto, PortalPopulateContext context, boolean overwrite) {
         return portalParticipantUserService.create(popDto);
     }
+
+    public List<String> bulkPopulateParticipants(String portalShortcode, EnvironmentName envName, String studyShortcode, Integer numEnrollees) {
+        StudyPopulateContext context = new StudyPopulateContext("portals/" + portalShortcode + "/participants/seed.json", portalShortcode, studyShortcode, envName, new HashMap<>());
+
+        List<String> populatedUsernames = new ArrayList<>();
+
+        IntStream.range(0, numEnrollees).forEach(i -> {
```
Comment (breilly2) [inline populate/src/main/java/bio/terra/pearl/populate/service/PortalParticipantUserPopulator.java]: Your FP is showing. 😂 While you're at it, may as well go all the way and use `collect` to construct your username list:
```
        List<String> populatedUsernames = IntStream.range(0, numEnrollees).mapToObj(i -> {
            try {
                ...
                return username;
            } catch (IOException e) {
                ...
            }
        }).collect(Collectors.toList());
```

**Review by breilly2 — APPROVED** [review]

File: populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java (hunk 1 above)
Comment (devonbush) [reply populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java:345]: yup, they might get halfway through filling out the consent form, so we'd have a saved partial response

File: populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java [hunk 3]
```diff
@@ -330,5 +333,25 @@ public Enrollee createNew(EnrolleePopDto popDto, StudyPopulateContext context, b
...
+                popDto.setSubmittedHoursAgo(PopulateUtils.randomInteger(0, 480)); //add some jitter to when they joined, to make graphs/views more interesting
```
Comment (devonbush) [inline populate/src/main/java/bio/terra/pearl/populate/service/EnrolleePopulator.java:346]: nice!

File: populate/src/main/resources/seed/portals/ourhealth/studies/ourheart/enrollees/seed.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
...AgQIECAAIFYAgJWrH6ohgABAgQIEGhAQMBqoImGQIAAAQIECMQSELBi9UM1BAgQIECAQAMCAlYDTTQEAgQIECBAIJaAgBWrH6ohQIAAAQIEGhAQsBpooiEQIECAAAECsQQErFj9UA0BAgQIECDQgICA1UATDYEAAQIECBCIJSBgxeqHaggQIECAAIEGBASsBppoCAQIECBAgEAsAQErVj9UQ4AAAQIECDQgIGA10ERDIECAAAECBGIJCFix+qEaAgQIECBAoAEBAauBJhoCAQIECBAgEEtAwIrVD9UQIECAAAECDQgIWA000RAIECBAgACBWAICVqx+qIYAAQIECBBoQEDAaqCJhkCAAAECBAjEEhCwYvVDNQQIECBAgEADAgJWA000BAIECBAgQCCWgIAVqx+qIUCAAAECBBoQELAaaKIhECBAgAABArEEBKxY/VANAQIECBAg0ICAgNVAEw2BAAECBAgQiCUgYMXqh2oIECBAgACBBgQErAaaaAgECBAgQIBALAEBK1Y/VEOAAAECBAg0ICBgNdBEQyBAgAABAgRiCQhYsfqhGgIECBAgQKABAQGrgSYaAgECBAgQIBBLQMCK1Q/VECBAgAABAg0ICFgNNNEQCBAgQIAAgVgCAlasfqiGAAECBAgQaEBAwGqgiYZAgAABAgQIxBIQsGL1QzUECBAgQIBAAwICVgNNNAQCBAgQIEAgloCAFasfqiFAgAABAgQaEBCwGmiiIRAgQIAAAQKxBASsWP1QDQECBAgQINCAgIDVQBMNgQABAgQIEIglIGDF6odqCBAgQIAAgQYEBKwGmmgIBAgQIECAQCwBAStWP1RDgAABAgQINCAgYDXQREMgQIAAAQIEYgkIWLH6oRoCBAgQIECgAQEBq4EmGgIBAgQIECAQS0DAitUP1RAgQIAAAQINCAhYDTTREAgQIECAAIFYAgJWrH6ohgABAgQIEGhAQMBqoImGQIAAAQIECMQSELBi9UM1BAgQIECAQAMCAlYDTTQEAgQIECBAIJaAgBWrH6ohQIAAAQIEGhAQsBpooiEQIECAAAECsQQErFj9UA0BAgQIECDQgICA1UATDYEAAQIECBCIJSBgxeqHaggQIECAAIEGBASsBppoCAQIECBAgEAsAQErVj9UQ4AAAQIECDQg8P/3ZrhBnabaAQAAAABJRU5ErkJggg=="
```
Comment (devonbush) [inline populate/src/main/resources/seed/portals/ourhealth/studies/ourheart/enrollees/seed.json]: to save space and readability (and possibly reduce insertion time), you could replace this with a smaller image (e.g. a base64 encoding of a small icon or something)

**Review by devonbush — APPROVED** [review]
Good stuff, I really like the approach you took with the seed files.  One suggestion, that's worth doing if it takes you < 5mins, otherwise don't worry about it

File: populate/src/main/resources/seed/portals/ourhealth/studies/ourheart/enrollees/seed.json (hunk 4 above)
Comment (MatthewBemis) [reply populate/src/main/resources/seed/portals/ourhealth/studies/ourheart/enrollees/seed.json]: Reduced by ~90% in c6bad97a26f9d0068f8898777a64022ec0c9c28d

File: populate/src/main/java/bio/terra/pearl/populate/service/PortalParticipantUserPopulator.java (hunk 2 above)
Comment (MatthewBemis) [reply populate/src/main/java/bio/terra/pearl/populate/service/PortalParticipantUserPopulator.java]: Good call, done in c6bad97a26f9d0068f8898777a64022ec0c9c28d

**Pull request merged**